### PR TITLE
test: unit test intacct import settings initialization

### PR DESCRIPTION
**Pull request by JustARatherRidiculouslyLongUsername**
## Clickup
https://app.clickup.com/t/86cwh86d6

## Summary by CodeRabbit

- **Tests**
	- Enhanced test suite for `IntacctImportSettingsComponent` to improve testing of service interactions and component behavior.
	- Added new test cases for form initialization and dependent field handling.
	- Improved mock data structures in `intacct.fixture.ts` to support expanded testing scenarios and configurations, including new constants for import settings and field mappings.

### CODE REVIEW

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,242 @@
...
+        { provide: SiImportSettingService, useValue: siImportSettingServiceSpy },
+        { provide: SiMappingsService, useValue: siMappingsServiceSpy },
+        { provide: IntacctConnectorService, useValue: intacctConnectorServiceSpy },
+        { provide: OrgService, useValue: orgServiceSpy },
+        { provide: TrackingService, useValue: trackingServiceSpy },
+        { provide: IntegrationsToastService, useValue: toastServiceSpy },
+        { provide: StorageService, useValue: storageServiceSpy },
+        { provide: SiWorkspaceService, useValue: siWorkspaceServiceSpy },
+        provideRouter([])
+        // HelperService
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:65]: remove it

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 2]
```diff
@@ -1,23 +1,242 @@
...
+      expect(siMappingsService.getConfiguration).toHaveBeenCalled();
+      expect(intacctConnectorService.getLocationEntityMapping).toHaveBeenCalled();
+      expect(siImportSettingService.getImportCodeFieldConfig).toHaveBeenCalled();
+    });
+
+    it('should correctly transform and set sageIntacctFields', () => {
+      expect(component.sageIntacctFields).toEqual([
+        {
+          attribute_type: 'PROJECT',
+          display_name: 'Project'
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:126]: move to fixture and get it from there

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 3]
```diff
@@ -1,23 +1,242 @@
...
+          expect(control.get('source_placeholder')).toBeTruthy();
+        };
+        expenseFieldsArray.controls.forEach(testForFields);
+      });
+    });
+
+    describe('Dependent Fields Setup', () => {
+      it('should handle dependent fields when project mapping exists', fakeAsync(() => {
+        const importSettingsWithProject = {...importSettings, mapping_settings: [{
+          source_field: 'PROJECT',
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:200]: have a separate fixture for this

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 4]
```diff
@@ -1,23 +1,242 @@
...
+        tick();
+
+        expect(component.costCodeFieldOption.length).toBe(1);
+        expect(component.costTypeFieldOption.length).toBe(1);
+      }));
+
+      it('should handle dependent field settings', () => {
+        const settingsWithDependentFields = {...importSettings, dependent_field_settings: {
+          is_import_enabled: true,
+          cost_code_field_name: 'COST_CODE',
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:216]: same as above

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 5]
```diff
@@ -1,23 +1,242 @@
...
+        }} as ImportSettingGet;
+        siImportSettingService.getImportSettings.and.returnValue(of(settingsWithDependentFields));
+
+        component.ngOnInit();
+        fixture.detectChanges();
+
+        expect(component.importSettingsForm.get('isDependentImportEnabled')?.value).toBeTrue();
+        expect(component.importSettingsForm.get('costCodes')?.value).toEqual({
+          attribute_type: 'COST_CODE',
+          display_name: 'COST_CODE',
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:229]: check it from fixture const

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts [hunk 6]
```diff
@@ -1,23 +1,242 @@
...
+          source_placeholder: 'Enter Cost Code',
+          is_dependent: true
+        });
+        expect(component.importSettingsForm.get('costTypes')?.value).toEqual({
+          attribute_type: 'COST_TYPE',
+          display_name: 'COST_TYPE',
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:235]: same as above

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts (hunk 5 above)
Comment (JustARatherRidiculouslyLongUsername) [reply src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:203]: The property names differ between the fixture and costCode field. That was why I created a separate object to check the field value. Is that okay?


image
image

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts (hunk 6 above)
Comment (JustARatherRidiculouslyLongUsername) [reply src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:209]: same

File: src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/integrations/intacct/intacct-shared/intacct-import-settings/intacct-import-settings.component.spec.ts:203]: you can create a separate fixture const with the properties you have

**Review by DhaaraniCIT — APPROVED** [review]

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**